mmcv/ops/nms.py

```diff
@@ -15,13 +14,24 @@
 class NMSop(torch.autograd.Function):
 
     @staticmethod
-    def forward(ctx, bboxes, scores, iou_threshold, offset):
+    def forward(ctx, bboxes, scores, iou_threshold, offset, score_threshold,
+                max_num):
+        if torch.onnx.is_in_onnx_export():
```

Comment:
There may be too many `if torch.onnx.is_in_onnx_export()` and better to simplify this.
For example, we only pass score_threshold > 0 and max_num > 0 if torch.onnx.is_in_onnx_export().
Here we could replace with `if score_threshold > 0`

Reply:
Removed `torch.onnx.is_in_onnx_export()` from `forward` and added a workaround with `max_num` in `batchd_nms`:
https://github.com/open-mmlab/mmcv/blob/5c083b88e3c312e34a707cface506126036a390a/mmcv/ops/nms.py#L310-L323

This should solve the problem with the loss of `bboxes` during training.

Reply:
In addition, we can now remove `dets = dets [: max_num]` after calling `batch_nms`, as, for example, in `mmdetection::multiclass_nms`:
https://github.com/open-mmlab/mmdetection/blob/2313bd76940a6059b7e8bad4908c4558b3160f75/mmdet/core/post_processing/bbox_nms.py#L84-L88

Reply:
It should be feasible for now.